packages/ketcher-core/src/application/editor/actions/template.ts

```diff
@@ -277,11 +334,27 @@ function fromTemplateOnBond(restruct, template, bid, flip) {
         tBond
       ).perform(restruct) as BondAdd
       action.addOp(operation)
-
-      pasteItems.bonds.push(operation.data.bid)
+      const newBondId = operation.data.bid
+
+      if (isFusingBenzeneBySpecialRules) {
+        const newBondType = benzeneDoubleBondIndexes.includes(tBondIndex)
+          ? Bond.PATTERN.TYPE.DOUBLE
+          : Bond.PATTERN.TYPE.SINGLE
+        action.addOp(
+          new BondAttr(newBondId, 'type', newBondType).perform(restruct)
+        )
+      }
+
+      pasteItems.bonds.push(newBondId)
     } else {
       const commonBond = bond.type > tmplBond.type ? bond : tmplBond
       action.mergeWith(fromBondsAttrs(restruct, existId, commonBond, true))
+
+      if (isFusingBenzeneBySpecialRules && benzeneConnectingBondType) {
+        action.addOp(
+          new BondAttr(bid, 'type', benzeneConnectingBondType).perform(restruct)
+        )
```

Comment:
Same here

```
this.addOption(bid, '', benzeneConnectingBondType)
```

Reply:
action.addOp actually means "add operation" :)
I just think that that method belongs in Action class and dont see much benefit in creating a local method that will basically just call the other one.